### PR TITLE
Feature/6 add login page

### LINKED ISSUE #6: Add Login page
**Description**
As a user
I want to log in and out of the application to edit my data.

**To-Do**
- [x] Create a log in page to log in via github
- [x] Protect front end routes so they can't be reachend unless logged in
- [x] Protect backend endpoints
- [x] Add a visual that indicates you're logged in
- [x] Add a logout button
- [x] Add tests for backend

### PR DESCRIPTION
**Pull request by jonashonecker**
**Description**
* Added spring security configuration
* Added an endpoint to retrieve user information and check whether the user is logged in
* Added integration tests to test new endpoint
* Added front end login page and main page
* Added protected routes to enforce logging into the application
* Added github avatar icon to be shown upon being logged in to provide a visual cue for being logged in

Closes #6

### CODE REVIEW

**Comment by jonashonecker** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=jonashonecker_TicketScout_backend&pullRequest=14) **Quality Gate failed for 'backend'**
> 
> Failed conditions  [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=jonashonecker_TicketScout_backend&pullRequest=14&resolved=false&sinceLeakPeriod=true)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=jonashonecker_TicketScout_backend&pullRequest=14)

This is not critical at this stage since the only thing the user can be tricked into is retrieving publicly available information from their GitHub profile. However, I have created a ticket for this issue to be fixed when there is time: #15.

**Review by gcode-de — APPROVED** [review]

File: backend/src/test/java/com/github/jonashonecker/backend/user/UserControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class UserControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void getMe_whenUserNotLoggedIn_returnUserInformation() throws Exception {
```
Comment (gcode-de) [inline backend/src/test/java/com/github/jonashonecker/backend/user/UserControllerTest.java]: maybe name this "returnUnauthorized"

**Review by gcode-de — APPROVED** [review]
nice work!

File: backend/src/test/java/com/github/jonashonecker/backend/user/UserControllerTest.java (hunk 1 above)
Comment (jonashonecker) [reply backend/src/test/java/com/github/jonashonecker/backend/user/UserControllerTest.java]: Good catch, thx!!!

**Pull request merged**